src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -1077,8 +1077,119 @@ def update_policy_setdefinition(cmd, policy_set_definition_name, definitions=Non
     return policy_client.policy_set_definitions.create_or_update(policy_set_definition_name, parameters)
 
 
+def _register_rp(cli_ctx, subscription_id=None):
+    rp = "Microsoft.Management"
+    from azure.cli.core.commands.client_factory import get_mgmt_service_client
+    from azure.cli.core.profiles import ResourceType
+    import time
+    rcf = get_mgmt_service_client(
+        cli_ctx,
+        ResourceType.MGMT_RESOURCE_RESOURCES,
+        subscription_id)
+    rcf.providers.register(rp)
+    while True:
+        time.sleep(10)
+        rp_info = rcf.providers.get(rp)
+        if rp_info.registration_state == 'Registered':
+            break
+
+
+def _get_subscription_id_from_subscription(cli_ctx, subscription):  # pylint: disable=inconsistent-return-statements
+    from azure.cli.core._profile import Profile
+    profile = Profile(cli_ctx=cli_ctx)
+    subscriptions_list = profile.load_cached_subscriptions()
+    for sub in subscriptions_list:
+        if sub['id'] == subscription or sub['name'] == subscription:
+            return sub['id']
+    from azure.cli.core.util import CLIError
+    raise CLIError("Subscription not found in the current context.")
+
+def _get_parent_id_from_parent(parent):
+    if parent is None or parent.startswith("/providers/Microsoft.Management/managementGroups/"):
+        return parent
+    return "/providers/Microsoft.Management/managementGroups/"+parent
+
+def cli_managementgroups_group_list(cmd, client):
+    _register_rp(cmd.cli_ctx)
+    return client.list()
+
+
+def cli_managementgroups_group_show(
+        cmd,
+        client,
+        group_name,
+        expand=False,
+        recurse=False):
+    _register_rp(cmd.cli_ctx)
+    if expand:
+        return client.get(group_name, "children", recurse)
+    return client.get(group_name)
+
+
+def cli_managementgroups_group_create(
+        cmd,
+        client,
+        group_name,
+        display_name=None,
+        parent=None):
+    _register_rp(cmd.cli_ctx)
+    parent_id=_get_parent_id_from_parent(parent)
+    from azure.mgmt.managementgroups.models import (CreateManagementGroupRequest, CreateManagementGroupDetails, CreateParentGroupInfo)
+    create_parent_grp_info = CreateParentGroupInfo(id=parent_id)
+    create_mgmt_grp_details = CreateManagementGroupDetails(parent=create_parent_grp_info)
+    create_mgmt_grp_request = CreateManagementGroupRequest(name=group_name, display_name=display_name, details=create_mgmt_grp_details)
+    return client.create_or_update(group_name, create_mgmt_grp_request)
+
+
+def cli_managementgroups_group_update_custom_func(
+        instance,
+        display_name=None,
+        parent=None):
+    instance["display_name"] = display_name
+    instance["parent"] = parent
+    return instance
```

Comment:
If the instance is the object, these simply become `instance.foo = foo`.  A little cleaner.